File: common/src/main/java/de/werkelmann/rekote/address/url/UrlValidCharRule.java
package de.werkelmann.rekote.address.url;

import de.werkelmann.rekote.address.ConnectionRule;

public class UrlValidCharRule implements ConnectionRule {
    @Override
    public boolean isValid(String address) {
        for (char c : address.toCharArray()) {
            if (!Character.isLetter(c) && !Character.isDigit(c) && !(c == '-') && !(c == '.')) {
                return false;
            }
        }
        return true;
    }
}

File: common/src/main/kotlin/de/werkelmann/rekote/address/url/UrlValidCharRule.kt
package de.werkelmann.rekote.address.url

import de.werkelmann.rekote.address.ConnectionRule

class UrlValidCharRule : ConnectionRule {
    override fun isValid(address: String): Boolean {
        address.toCharArray().forEach {
            if (!Character.isLetter(it) && !Character.isDigit(it) && it != '-' && it != '.') {
                return false
            }
        }
        return true
    }
}
